### PR TITLE
Add: Block.addRunningEffects

### PR DESCRIPTION
**Pull request by ShetiPhian**
Adds Block.addRunningEffects to go with the existing ones enabling a mod to work around the vanilla particles using the wrong blockstate.

### CODE REVIEW

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 1]
```diff
@@ -827,6 +827,20 @@
 +    }
 +
 +    /**
++     * Allows a block to override the standard Entity.createRunningParticles particles.
++     *
++     * @param worldObj The current world
++     * @param blockpos of the block that the entity is running on.
++     * @param iblockstate State at the specific world/pos
++     * @param entity the entity that hit landed on the block.
```
Comment (williewillus) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: Skipped updating this Javadoc? Should be "entity that is running on the block"

Comment (ShetiPhian) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: Thought I got them all, it was 4am though.

**Comment by Alex-the-666** [comment]
Could you make it so that it could somehow be cancel able? for example, if the player is changed to the size of a bug it still makes large particles. If I could cancel this I could spawn my own, smaller particles. I know it's not an event, but it would be nice.

**Comment by ShetiPhian** [comment]
You just need to return true to cancel the vanilla spawn.

**Comment by Alex-the-666** [comment]
Ah, I'll probably do that.

**Pull request closed without merging**